runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -420,22 +301,52 @@ public void initChannel(SocketChannel ch) throws Exception {
         }
     }
 
-    synchronized void connectChannel(Bootstrap b, long c) {
+    /** Get the {@link ChannelInitializer} used for initializing the Netty channel pipeline.
+     *
+     * @return A {@link ChannelInitializer} which initializes the pipeline.
+     */
+    private ChannelInitializer getChannelInitializer() {
+        return new ChannelInitializer() {
+            @Override
+            protected void initChannel(@Nonnull Channel ch) throws Exception {
+                if (parameters.isTlsEnabled()) {
+                    ch.pipeline().addLast("ssl", sslContext.newHandler(ch.alloc()));
+                }
+                ch.pipeline().addLast(new LengthFieldPrepender(4));
+                ch.pipeline().addLast(new LengthFieldBasedFrameDecoder(Integer.MAX_VALUE,
+                        0, 4, 0,
+                        4));
+                if (parameters.isSaslPlainTextEnabled()) {
+                    PlainTextSaslNettyClient saslNettyClient =
+                        SaslUtils.enableSaslPlainText(parameters.getUsernameFile(),
+                            parameters.getPasswordFile());
+                    ch.pipeline().addLast("sasl/plain-text", saslNettyClient);
+                }
+                ch.pipeline().addLast(new NettyCorfuMessageDecoder());
+                ch.pipeline().addLast(new NettyCorfuMessageEncoder());
+                ch.pipeline().addLast(new ClientHandshakeHandler(parameters.getClientId(),
+                        node.getNodeId(), parameters.getHandshakeTimeout()));
+                ch.pipeline().addLast(NettyClientRouter.this);
+            }
+        };
+    }
+
+    private synchronized void connectChannel(Bootstrap b)
+            throws ConnectException, NoRouteToHostException, SocketException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.net.ConnectException' which is a subclass of 'java.net.SocketException'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.net.NoRouteToHostException' which is a subclass of 'java.net.SocketException'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.net.SocketException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)